src/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs

```diff
@@ -0,0 +1,117 @@
+        public static EventWaitHandle Create(bool initialState, EventResetMode mode, string name, out bool createdNew, EventWaitHandleSecurity eventSecurity)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_CannotBeNullOrEmpty), nameof(name));
+            }
+            if (name.Length > Interop.Kernel32.MAX_PATH)
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, name), nameof(name));
+            }
+            if (eventSecurity == null)
+            {
+                throw new ArgumentNullException(nameof(eventSecurity));
+            }
+
+            bool isManualReset = mode switch
+            {
+                EventResetMode.ManualReset => true,
+                EventResetMode.AutoReset => false,
+                _ => throw new ArgumentOutOfRangeException(nameof(mode), SR.Format(SR.ArgumentOutOfRange_Enum))
+            };
+
+            return CreateEventWaitHandle(initialState, isManualReset, name, out createdNew, eventSecurity);
+        }
+
+        private static unsafe EventWaitHandle CreateEventWaitHandle(bool initialState, bool isManualReset, string name, out bool createdNew, EventWaitHandleSecurity security)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name));
+            Debug.Assert(security != null);
+
+            EventWaitHandle eventHandle = null;
+
+            fixed (byte* pSecurityDescriptor = security.GetSecurityDescriptorBinaryForm())
+            {
+                var secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
+                {
+                    nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES),
+                    lpSecurityDescriptor = (IntPtr)pSecurityDescriptor
+                };
+
+                SafeWaitHandle handle = Interop.Kernel32.CreateEvent(ref secAttrs, isManualReset, initialState, name);
+                ValidateHandle(handle, name, out createdNew);
+
+                AggregateException aggEx = null;
+
+                // Ensure we always close the handle regardless of the OpenExisting call result
+                try
+                {
+                    eventHandle = EventWaitHandle.OpenExisting(name);
+                }
+                catch (Exception ex)
+                {
+                    aggEx = new AggregateException(ex);
+                }
+                finally
+                {
+                    handle.Dispose();
+                }
+
+                if (aggEx != null)
+                {
+                    throw aggEx;
+                }
+            }
+
+            return eventHandle;
+        }
+
+        private static void ValidateHandle(SafeWaitHandle handle, string name, out bool createdNew)
+        {
+            Debug.Assert(handle != null);
```

Comment:
This assert does not seem worth having since it is checked (by not throwing) moments later on line 104. In C#, we do not use Debug.Assert to check for null before dereferencing. We use it to check for bad states that otherwise would not be detected until some time later in the program when the problem would be harder to figure out.

Reply:
Ok thanks. Removing it.